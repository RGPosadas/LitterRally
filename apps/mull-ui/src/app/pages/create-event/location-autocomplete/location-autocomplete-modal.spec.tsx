import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';

import LocationAutocompleteModal from './location-autocomplete-modal';

const mockFormik = {
  values: { location: 'test string' },
  touched: { location: false },
  errors: { location: false },
};

describe('location autocomplete modal', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<LocationAutocompleteModal formik={mockFormik} />);
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const tree = renderer.create(<LocationAutocompleteModal formik={mockFormik} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should open autocomplete modal', async () => {
    const utils = render(<LocationAutocompleteModal formik={mockFormik} />);
    const autocompleteModal = utils.container.querySelector('#location');
    await waitFor(() => {
      fireEvent.click(autocompleteModal);
    });
  });
});
