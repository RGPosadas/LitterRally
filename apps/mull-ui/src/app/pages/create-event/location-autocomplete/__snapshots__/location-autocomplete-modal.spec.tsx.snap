// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`location autocomplete modal should match snapshot 1`] = `
<div
  className="custom-text-input-container "
>
  <label
    className="custom-text-input-label"
    htmlFor="location"
  >
    Location
  </label>
  <div
    className="custom-text-input-sub-container"
  >
    <input
      className="custom-text-input input-border "
      data-testid="custom-text-input"
      id="location"
      name="location"
      onClick={[Function]}
      readOnly={true}
      type="text"
      value=""
    />
    <div
      className="custom-text-input-icon"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-map-marker-alt fa-w-12 input-icon"
        data-icon="map-marker-alt"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;
